bindings/fcl-js.go

```diff
@@ -0,0 +1,111 @@
+package bindings
+
+import (
+	"bytes"
+	"embed"
+	"sort"
+	"text/template"
+
+	"github.com/onflow/flixkit-go"
+	"github.com/stoewer/go-strcase"
+)
+
+type SimpleParameter struct {
+	Name        string
+	JsType      string
+	Description string
+	FclType     string
+	CadType     string
+}
+
+type TemplateData struct {
+	Version         string
+	Parameters      []SimpleParameter
+	Title           string
+	Description     string
+	Location        string
+	IsScript        bool
+	IsLocalTemplate bool
+}
+
+//go:embed templates/*.tmpl
+var templateFiles embed.FS
```

Comment:
If this library is to be used by a dependency I am not sure this works. Have we tried sending in a fs.FS with the files when we create the generator on line 34 here instead, that scales tons better across composition and tests as well.

Reply:
good point, this way any caller can provide their own templates. updated PR.